juju/application.py

```diff
@@ -312,23 +312,40 @@ def collect_metrics(self):
             [],
         )
 
+    async def get_annotations(self):
```

Comment:
To reduce the repetition of these methods, maybe we should factor them up to a `AnnotatableEntity` mixin class?  I was thinking it could just be moved up to the `ModelEntity` base, but `Relation` is also a `ModelEntity` and maybe isn't annotatable (but should it be)?

The `Annotation` class is also a `ModelEntity` but I think we should just remove that entirely, since it is unused and overkill.  Would you mind doing that in this PR?

Reply:
I just used some stand alone functions. The model had connection as a function whereas machine, unit, and application had a property.

I also wasn't clear that the mixin could reach down into attributes of the derived class in order to get acces to tag and connection.

Reply:
A mixin could use other attributes on the derived class as long as it documented that as part of the required interface, but I think the helper functions are just as good, just a different style.

Reply:
Also, I dislike the inconsistency between the various `connection` properties / methods, but that was an oversight in the macaroon bakery work.  Not sure if we could reasonably clean it up at this point.